crates/core/protobuf/node_ctl_svc.proto

```diff
@@ -34,9 +34,9 @@ message ProvisionClusterRequest {
   bool dry_run = 1;
   // if unset then the configured cluster num partitions will be used
   optional uint32 num_partitions = 2;
-  // if unset partition replication will default to `everywhere`. Otherwise
-  // it's limited to the provided replication property
-  optional restate.cluster.ReplicationProperty partition_replication = 3;
+  // if unset partition replication will default to the configured
+  // `default-partition-replication`
+  optional restate.cluster.PartitionReplication partition_replication = 3;
```

Comment:
Do we need to keep the old field for ensuring backward compatibility with older restatectl versions? Would a new `restatectl` version work with an older server?